manifests/repository.pp

```diff
@@ -4,10 +4,14 @@
       include ::apt
       apt::source { 'puppetlabs':
         location   => 'http://apt.puppetlabs.com',
-        repos      => 'main',
-        key        => '4BD6EC30',
-        key_server => 'pgp.mit.edu',
+        repos      => 'main dependencies',
+        key        => {
+          id     => '47B320EB4C7C375AA9DAE1A01054B7A24BD6EC30',
+          server => 'hkp://pgp.mit.edu:80',
+        },
+        notify     => Exec['apt_update'],
       }
+      Class['apt::update'] -> Package <| |>
```

Comment:
I'm with you but without this, the notify is not working for me. I'm using Puppet 3.8.5 with puppetlabs-apt 2.2.1.